### ISSUE TITLE
Feature Request for Reverse Type Cloze Cards

### ISSUE THREAD

**Issue by abhishek7786** [Vilhelm-Ian/Interactive_And_Randomize_Anki_Note_Types#1]
@Vilhelm-Ian Thanks for these new cloze cards.

One of the issue with cloze cards is it doesn't support functionality of front and back card.

For eg., ->
Card ->
Content.......{{cloze1}}.....content {{cloze 2}} ......content.....{{cloze:: Answer :: Hint/Question }}.....content...

So now it creates 3 cards -> First card for cloze 1; second card for cloze 2 and third one for cloze::Answer with Hint/Question displayed initially.

Now if user want to test third cloze in revere manner that is cloze::Hint/Question::Answer then one more card needs to be created this adds the efforts and creates duplicity of content.

If there is way to distinguish the third cloze which creates two cards, one as {{Cloze::Answer::Hint/Question}} and second as {{cloze::Hint/Question::Answer}} then it reduces the burden. ( may be we can use {{RC1::question::Answer}} - RC - ReverseClzoe.

Example from real life to describe the issue ->

Note -> 
{{RC1::Historical Materialism is a framework developed by ? :: which Framework developed by Karl Marx ? }}  to understand the development of human societies. 
Marx contended that {{RC2::Material conditions, rather than ideas derive what ? :: What drives historical change ? }} 

It creates  4 cards.
Card 1 -> 
{{Framework developed by Karl Marx ? }} to understand the development of human Societies.
Marx Contended that {{}}

Card 2 ->
{{Historical Materialism is a framework developed by ? }} to understand the development of human societies. 
Marx contended that {{}}

Card 3 ->
{{}} to understand the development of human Societies.
Marx contended that {{what derives historical change ? }}

Card 4 ->
{{}} to understand the development of human Societies.
Marx contended that {{ Material conditions, rather than ideas derive what ? }}

Additional Info :- This request has been on anki platform for long attaching few links for the reference. 

https://forums.ankiweb.net/t/combine-basic-and-optional-reversed-and-cloze-notetypes-into-one-standard-card-type/27184
https://forums.ankiweb.net/t/reverse-cards-for-cloze-card-type-for-windows/15714
https://www.reddit.com/r/Anki/comments/oy8z3j/how_to_create_a_reversed_cloze_card/
https://stackoverflow.com/questions/48170984/how-to-implement-reverse-cloze-in-anki-2-1-x

**Comment by Vilhelm-Ian** [comment]
I think I got it. No problem I can create such a note type. But can you design. Gimme a list of features it should contain.

I will start.

1. It should support both basic and cloze
2. It should suppport reversed.

anything else

**Comment by Vilhelm-Ian** [comment]
Acutually you can achieve that already with the random cloze card.

Just do this. 

What is the meaning of hund {{c1::?::}} | What is dog in german {{c2::?::} | A domesticated animal that cases cats {{c3::Dog}}

**Comment by abhishek7786** [comment]
@Vilhelm-Ian I tried the below approach you suggest  - 
What is the meaning of hund {{c1::?::}} | What is dog in german {{c2::?::} | A domesticated animal that cases cats {{c3::Dog}}

It has the same issue for which reverse card is needed ( Above suggested approach is for removing the pattern matching issue in anki ) but it requires the effort to create more than one cards although with use of '|' symbol, also it treats those cards as single card, where as with reverse cloze card objective is to have two separate cards which test user same info in multiple ways. ( e.g, -> Card 1 -> Das Capital book written by {{}} and card 2 ->  {{}} is written by Karl Marx.

What I suggested above is to reduce the efforts in creating cards.

I'm putting the feature list here ->

1. It should support both basic and cloze
2. It should support reversed.
3. It should support the nested clozes.
4. An extra field, its content should be visible upon pressing 'show answer'
5. Configuration at card level 
   5.1. To turn on/off hint on clozes.
   5.2. To show/hide other clozes present in the note
   5.3. To configure how many clozes before and how many clozes after should be visible
   5.4. Highlight or change color of the cloze being front and back.
 
Basically functionality of reverse cloze card in these cloze cards. - https://github.com/nihil-admirari/modern-cloze-overlapper.

**Comment by Vilhelm-Ian** [comment]
@abhishek7786 If you want it to be treated at as two separate cards just change the cloze index?

{{c2::Das Capital}} book written by {{c1::Karl Mark}} | {{c2::Das Kapital}} is written by {{c1::Karl Marx}}
